tests/checkpointing/test_model_checkpoint.py

```diff
@@ -1317,3 +1317,39 @@ def test_trainer_checkpoint_callback_bool(tmpdir):
     mc = ModelCheckpoint(dirpath=tmpdir)
     with pytest.raises(MisconfigurationException, match="Invalid type provided for checkpoint_callback"):
         Trainer(checkpoint_callback=mc)
+
+
+@pytest.mark.parametrize("num_sanity_val_steps", [0, 2])
+@pytest.mark.parametrize("every_train_step", [True, False])
+def test_invalid_monitor(tmpdir, every_train_step, num_sanity_val_steps):
+
+    class LoggingModel(BoringModel):
+
+        def training_step(self, batch, batch_idx):
+            self.log("train_monitor", 0.1)
+            return super().training_step(batch, batch_idx)
+
+        def validation_step(self, batch, batch_idx):
+            self.log("val_monitor", 0.2)
+            return super().validation_step(batch, batch_idx)
+
+    mc = ModelCheckpoint(
+        dirpath=tmpdir,
+        monitor="invalid",
+        every_n_train_steps=every_train_step,
+        every_n_val_epochs=(not every_train_step),
+    )
+    model = LoggingModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        logger=False,
+        max_steps=2,
+        num_sanity_val_steps=num_sanity_val_steps,
+        callbacks=[mc],
+    )
+    with pytest.raises(MisconfigurationException, match=r"ModelCheckpoint\(monitor='invalid'\) not found"):
```

Comment:
I am not sure to understand this test. It shouldn't raise an exception if validation hasn't been run.

Reply:
this is a draft. 
I'm trying to undestand why _has_run was introduced.